### PR TITLE
Fix GetAllMods method

### PR DESCRIPTION
**Pull request by flibber-hk**

### CODE REVIEW

**Comment by SFGrenade** [comment]
here we see a description like "make the method do what the comment says".

**Comment by flibber-hk** [comment]
If you prefer, I could have changed the comment to something like

        /// <param name="onlyEnabled">Should the iterator exclude enabled mods, despite the parameter name</param>
        /// <param name="allowLoadError">Should the iterator exclude contain mods which don't have load errors, despite the parameter name</param>

File: Assembly-CSharp/ModHooks.cs [hunk 1]
```diff
@@ -2190,7 +2190,7 @@ Type type
             bool onlyEnabled = false,
             bool allowLoadError = false
         ) => ModLoader.ModInstances
-            .Where(x => onlyEnabled && !x.Enabled || !allowLoadError && x.Error != null)
+            .Where(x => !(onlyEnabled && !x.Enabled || !allowLoadError && x.Error != null))
```
Comment (fifty-six) [inline Assembly-CSharp/ModHooks.cs:2193]: Simplify the conditional to like `(x.Enabled || !onlyEnabled) && (x.Error != null || allowLoadError)` or `(!onlyEnabled || x.Enabled) && (allowLoadError || x.Error != null)` please. The nested nots are hard to read.

**Review by fifty-six — CHANGES_REQUESTED** [review]

**Comment by fifty-six** [comment]
If you want, you could change the parameter name to something like `allowDisabled` and then use `allowDisabled` instead of `!onlyEnabled`, because it doesn't seem like this is in-use anywhere as far as I can tell from a cursory github search. Up to you though.

**Comment by flibber-hk** [comment]
I was copying the style from the GetMod methods, so I changed those too in the latest commit. Parameter name `allowDisabled` is probably slightly better but I don't really feel it's worth changing (I don't have a strong opinion on this though)

**Pull request merged**